HotelManagerment/com.spring.sample/src/com/spring/sample/controller/RoomCategoriesController.java

```diff
@@ -29,12 +32,72 @@ public class RoomCategoriesController {
 
 	@GetMapping
 	public String index(Locale locale, Model model) {
-		// In toan bo danh sach phong
 		List<Room> listRooms = roomService.getAllRooms();
 		model.addAttribute("listRooms", listRooms);
-		// Hien thi danh sach cac loai phong
 		List<RoomType> listTypeOfRooms = roomTypeService.getAllTypeRooms();
 		model.addAttribute("listTypeRooms", listTypeOfRooms);
 		return "RoomCategory/index";
 	}
-}
\ No newline at end of file
+
+	// XÓA PHÒNG
+	@PostMapping("/delete")
+	public String deleteRoom(@RequestParam("roomID") String roomID, Model model) {
+
+		try {
+
+			roomService.deleteRoomById(roomID);
+			logger.info("Phòng với ID: {} đã được xóa thành công.", roomID);
+
+		} catch (Exception e) {
+
+			model.addAttribute("errorMessage", "Có lỗi xảy ra khi xóa phòng với ID: " + roomID + ". Vui lòng thử lại.");
+		}
+
+		return "redirect:/room-categories";
+	}
+
+	// Cập nhật thông tin phòng
+	@PostMapping("/update")
+	public String updateRoom(@RequestParam("roomID") String roomID, @RequestParam("nameRoom") String nameRoom,
+			@RequestParam("typeRoom") String typeRoomId, @RequestParam("noteRoom") String note) {
+		System.out.println("Bắt đầu update ");
+		// In các giá trị đầu vào
+		System.out.println("roomID: " + roomID);
+		System.out.println("nameRoom: " + nameRoom);
+		System.out.println("typeRoomId: " + typeRoomId);
+		System.out.println("noteRoom: " + note);
+		// Tìm RoomType bằng ID
+		RoomType roomType = roomTypeService.findById(typeRoomId);
+		// Cập nhật phòng
+		roomService.UpdateRoom(roomID, nameRoom, note, roomType);
```

Comment:
check xem update ko thành công haowjc thành công thì có mess j ko nhá
tên method bắt đầu với chữ cái thường chư sko phải in hoa